test/framework/clusterctl/clusterctl_helpers.go

```diff
@@ -272,3 +276,25 @@ func ApplyClusterTemplateAndWait(ctx context.Context, input ApplyClusterTemplate
 		Cluster: result.Cluster,
 	}, input.WaitForMachineDeployments...)
 }
+
+// setDefaults sets the default values for ApplyClusterTemplateAndWaitInput if not set.
```

Comment:
nit: Just add a comment reporting that the current implementation of the default waiters are designed for a KCP control plane.